HW0/P4/P4.py

```diff
@@ -30,47 +51,105 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    # Read the file
+    trueFile = open("P4_trajectory.txt","r"); 
+    x_coords, y_coords, z_coords = np.loadtxt(trueFile,delimiter =',',usecols = (0,1,2),unpack = True);
+    ax.plot(x_coords, y_coords, z_coords,
+             '--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    # Read the file
+    ObsFile = open("P4_measurements.txt","r");
+    x,y,z = np.loadtxt(ObsFile, delimiter = ',', unpack = True);
+    # Constract Measurement array 
+    M = np.array([x,y,z]);
+    x_coords = (1/rx) * M[0];
+    y_coords = (1/ry) * M[1];
+    z_coords = (1/rz) * M[2];    
+    ax.plot(x_coords, y_coords, z_coords,
+             '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
-
+    A = np.matrix([[1,0,0,dt,0,0],[0,1,0,0,dt,0],[0,0,1,0,0,dt],[0,0,0,(1-c*dt),0,0],[0,0,0,0,(1-c*dt),0],[0,0,0,0,0,(1-c*dt)]]);
+    a = np.matrix([[0,0,0,0,0,(g*dt)]]);
+    a = a.transpose();
+    s0 = np.matrix([[0,0,2,15,3.5,4.0]])
+    s0 = s0.transpose();
+    #Construct a 6 * K matrix
+    S = np.empty(shape=(6,K), dtype = float)
+    # Fill in all the columns 
     # Initial conditions for s0
+    S[:,0] = s0.transpose();
+    # Set previous array equals to 0
+    s_p = s0;
     # Compute the rest of sk using Eq (1)
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '-k', label='Blind trajectory')
+    for x in range(1,K):
+        s_k = A * s_p + a;
+        S[:,x] = s_k.transpose();
+        s_p = s_k;
+
+    # Filled all 6*K matrix
+    # Plot the first three row in the array
+    x_coords = S[0,:];
+    y_coords = S[1,:];
+    z_coords = S[2,:];
+     
+    # Print x_coords, y_coords, z_coords
+    ax.plot(x_coords, y_coords, z_coords,
+             '-k', label='Blind trajectory')
 
     #####################
     # Part 4:
     # Use the Kalman filter for prediction
     #####################
-
-    # B = ?
-    # C = ?
+    # Create the B Matrix
+    B = np.identity(6);
+    b = np.matrix([[bx,by,bz,bvx,bvy,bvz]]);
+    B = b * B;
```

Comment:
It seems like B is a vector and not a matrix. This has an impact on the computation of the kalman filter solution.